dot-studio/svc_coord

```diff
@@ -51,42 +51,75 @@ function wait_for_svc_to_load() {
 }
 
 document "wait_for_port_to_listen" <<DOC
-  Wait for a port to be listening.
-
+  Wait for a port to be listening. If the port is not found in the set amount of time the function exits with 1
   @(arg:1) Port to wait for to be listening
-  @(arg:2) Wheter or not this process runs silently
+  @(arg:2) Number of seconds to wait to exit with 1. Default 60 seconds.
+  @(arg:3) Wheter or not this process runs silently
+
+  Example: Wait for a service to start listening on port 1234 for 1 minute
+  ---------------------------------------------
+  wait_or_fail_for_port_to_listen 1234 60
 DOC
 function wait_for_port_to_listen() {
   install_if_missing core/busybox-static netstat
+  SECOUNDS_WAITING=$2
+  if [ -z "$SECOUNDS_WAITING" ]
+  then
+    SECOUNDS_WAITING=60
+  fi
   
+  COUNTER=0
```

Comment:
This counter could be good to create it locally so we don't export it outside this function.
```bash
local COUNTER=0
```